### PR TITLE
Refactor and add tests for git tag parsing

### PR DESCRIPTION
**Pull request by cam72cam**
This breaks out the git stdout parsing logic into it's own function and add tests around it.  We could add a test for exec-git at some point, but it's a clean enough failure that I'm not going to worry about it.

### CODE REVIEW

File: src/internal/github/git_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func Test_parseTagsFromStdout(t *testing.T) {
+	cases := map[string]struct {
+		input       []string
+		expected    []string
+		expectedErr string
+	}{
```
Comment (kislerdm) [inline src/internal/github/git_test.go:14]:
```suggestion
	cases := map[string]struct {
		input       []string
		expected    []string
		expectedErr errror
	}{
```
...

File: src/internal/github/git_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+				if err != nil {
+					t.Errorf("Unexpected error: %v", err)
+				}
+			} else {
+				if err == nil {
+					t.Errorf("Expected error %s, got <nil>", tc.expectedErr)
+				} else if err.Error() != tc.expectedErr {
+					t.Errorf("Expected error %s, got %v", tc.expectedErr, err)
+				}
+			}
```
Comment (kislerdm) [inline src/internal/github/git_test.go:59]: I find this is easier to digest given that we use testify. WDYT?

```suggestion
if assert.ErrorIs(t, err, tc.expectedErr) {
    t.Fatalf("Unexpected error. Want: %v, got: %v", tc.expectedErr, err)
}
```

File: src/go.mod [hunk 3]
```diff
@@ -3,6 +3,7 @@ module registry-stable
 go 1.21
 
 require (
+	github.com/google/go-cmp v0.5.9
```
Comment (kislerdm) [inline src/go.mod:6]: **(nitpick)** I feel that we could use either cmp, or testify, both feels overhead. WDYT?

**Review by kislerdm — APPROVED** [review]
I left a couple of minor nitpicks, LGTM otherwise 👍🏻

File: src/internal/github/git.go [hunk 4]
```diff
@@ -8,41 +8,55 @@ import (
...
-	for _, line := range strings.Split(buf.String(), "\n") {
+	for _, line := range lines {
 		if !strings.Contains(line, "refs/tags/") {
 			continue
 		}
 
 		fields := strings.Fields(line)
 		if len(fields) != 2 {
-			return nil, fmt.Errorf("could not parse tags for %s: tags are in wrong format", repositoryUrl)
+			return nil, fmt.Errorf("Invalid format for tag '%s', expected two fields", line)
```
Comment (cube2222) [inline src/internal/github/git.go:22]: Generally, errors in Go shouldn't start with capital letters.

File: src/internal/github/git.go [hunk 5]
```diff
@@ -8,41 +8,55 @@ import (
...
-	cmd := exec.Command("git", "ls-remote", "--tags", "--refs", repositoryUrl)
-	cmd.Stdout = &buf
-	cmd.Stderr = &bufErr
-
-	if err := cmd.Run(); err != nil {
-		log.Printf("Could not get tags for repository %s: %s", repositoryUrl, bufErr.String())
-		return nil, err
-	}
+func parseTagsFromStdout(lines []string) ([]string, error) {
+	exec.LookPath("")
```
Comment (Yantrio) [inline src/internal/github/git.go:12]: What's this LookPath call doing here? 👀

File: src/internal/github/git.go [hunk 6]
```diff
@@ -8,41 +8,55 @@ import (
...
+	log.Printf("Getting tags for repository %s", repositoryUrl)
+
+	var buf bytes.Buffer
+	var bufErr bytes.Buffer
+	cmd := exec.Command("git", "ls-remote", "--tags", "--refs", repositoryUrl)
+	cmd.Stdout = &buf
+	cmd.Stderr = &bufErr
+
+	if err := cmd.Run(); err != nil {
+		log.Printf("Could not get tags for repository %s: %s", repositoryUrl, bufErr.String())
```
Comment (Yantrio) [inline src/internal/github/git.go:51]: Maybe this error should be a bit more specific, mainly around how executing the `git` command failed.

Maybe also reporting the exit statuscode too?

File: src/internal/github/git_test.go [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_parseTagsFromStdout(t *testing.T) {
+	cases := map[string]struct {
+		input       []string
+		expected    []string
+		expectedErr error
+	}{
+		"Simple Tag": {
```
Comment (Yantrio) [inline src/internal/github/git_test.go:16]: It would be cool if we could add tests for empty string inputs (for fresh repos)

File: src/internal/github/git.go [hunk 8]
```diff
@@ -8,41 +8,55 @@ import (
...
+		}
+
 		tags = append(tags, tag)
 	}
 
-	log.Printf("Found %d tags for repository %s", len(tags), repositoryUrl)
+	return tags, nil
+}
 
+func GetTags(repositoryUrl string) ([]string, error) {
```
Comment (Yantrio) [inline src/internal/github/git.go:41]: Preferably I'd like GoDoc comments for atleast each exported function from a package. Even if it seems self explanatory, every little helps new developers approaching this.

**Review by Yantrio — CHANGES_REQUESTED** [review]
Just a few nitpicks :) Otherwise it looks good, and I love the addition of the tests. Thanks!

File: src/internal/github/git.go (hunk 6 above)
Comment (cam72cam) [reply src/internal/github/git.go]: Updated to:
```
Could not get tags for www.creedthoughts.gov.wwwcreedthoughts: git failed (128) fatal: 'www.creedthoughts.gov.wwwcreedthoughts' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
```

**Review by Yantrio — APPROVED** [review]
